components/merino/curated_recommendations/src/http.rs

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use crate::error::Error;
+use crate::models::{CuratedRecommendationsRequest, CuratedRecommendationsResponse};
+use url::Url;
+use viaduct::{header_names, Request, Response};
+
+pub struct HttpClient {}
+
+impl HttpClient {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn make_curated_recommendation_request(
+        &self,
+        request: &CuratedRecommendationsRequest,
+    ) -> Result<CuratedRecommendationsResponse, Error> {
+        let url = Url::parse("https://merino.services.mozilla.com/api/v1/curated-recommendations")?;
+        log::trace!("making request: {url}");
+        let response: Response = Request::post(url)
+            .header(header_names::ACCEPT, "application/json")?
+            .header(
+                header_names::USER_AGENT,
+                "application-services merino client",
+            )?
+            .json(request)
+            .send()?;
+
+        let status = response.status;
+        if status >= 400 {
+            let error_message = response.text();
+            let error = match status {
+                400 => Error::BadRequest {
+                    code: status,
+                    message: error_message.parse().unwrap(),
```

Comment:
Do you need the `parse().unwrap()`?

 I think `error_message` should be a more-or-less a string -- maybe a `Cow`.  Something like `.to_string()`, `.to_owned`, or `.into()` should convert it to a `String`.